src/libguac/fips.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "config.h"
+#include "guacamole/fips.h"
+
+#include <stdio.h>
+
+#ifdef ENABLE_SSL
+#include <openssl/opensslv.h>
+
+/* OpenSSL versions prior to 0.9.7e did not have FIPS support */
+#if !defined(OPENSSL_VERSION_NUMBER) || (OPENSSL_VERSION_NUMBER < 0x00090705f)
+
+int guac_fips_enabled() {
+
+	return 0;
+
+}
+
+/* OpenSSL 3+ uses EVP_default_properties_is_fips_enabled() */
+#elif defined(OPENSSL_VERSION_MAJOR) && (OPENSSL_VERSION_MAJOR >= 3)
+
+#include <openssl/evp.h>
+
+int guac_fips_enabled() {;
```

Comment:
I'm repeating the function definition here because these different FIPS checks require different headers to be included, and this seemed like the most straightforward way to do it.

Reply:
Is that syntax correct, though, with the `;` at the end of the line?

Reply:
Definitely not! fixed.

Reply:
Let me test this again before I un-draft it.

Reply:
Ooooops, I somehow lost my Makefile changes. Definitely didn't work without those.